File: code/functions/rprism/R/load_colors.R

#' Load lists of colors from Excel table
#'
#' Load one or multiple sheets from an excel workbook containing lists of colors for categories of different fields.
#'
#' @return A named list or a \code{tibble} dataframe. Names are categories of some fields or custom names and values are
#' colors in hexadecimal character vectors.
#' @param sheet Name of the excel spreadsheet.
#' @param as_tibble (optional) Set to TRUE if you would like to have the list of colors returned in a \code{tibble}
#' dataframe. If FALSE, colors are returned as a named list.
#' @importFrom readxl read_excel
#'
#' @export
load_colors <- function(sheet, as_tibble=F){
  cwd <- setwd_to_results()
  filepath <- get_filepath_colors()
  df_colors <- read_excel(path=filepath, sheet=sheet)
  setwd(cwd)

  if (as_tibble){
    return(df_colors)
  } else {
    colors <- list()
    for (i in 1:nrow(df_colors)){
      colors[[df_colors[i, "Name", drop=T]]] <- df_colors[i, "Color", drop=T]
    }
    return(colors)
  }
}

File: code/functions/pyprism/pyprism/data/_load_colors.py
# -*- coding: utf-8 -*-
"""
@created: 05/19/21
@modified: 16/06/21
@author: Yoann Pradat

    CentraleSupelec
    MICS laboratory
    9 rue Juliot Curie, Gif-Sur-Yvette, 91190 France

Function for loading lists of colors.
"""

import os
import numpy as np
import pandas as pd

from pyprism.util import setwd_to_data
from ._filepaths import _get_filepath_colors

DataFrame = pd.core.frame.DataFrame


# User function ========================================================================================================

def load_colors(sheet: str, as_dataframe=False):
    """Load one or multiple sheets from an excel workbook containing lists of colors for categories of different fields.

    Parameters
    ----------
    sheet: str
        Name of the excel sheet to be loaded.
    as_dataframe: bool, optional
        Set to True if you would like to have the list of colors returned in a dataframe. If False, colors are returned
        as a dictionary

    Returns
    -------
    A dict or a data.frame.

    """
    try:
        filepath = _get_filepath_colors()
        cwd = setwd_to_data()
        df_colors = pd.read_excel(filepath, sheet_name=sheet, engine="openpyxl", keep_default_na=False)
        os.chdir(cwd)
    except Exception as e:
        raise e

    if as_dataframe:
        return df_colors
    else:
        colors = dict(zip(df_colors["Name"], df_colors["Color"]))
        return colors
